client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -408,6 +410,9 @@ private String toTimestampLiteral(Object value)
         if (value instanceof LocalDateTime) {
             return LOCAL_DATE_TIME_FORMATTER.format(((LocalDateTime) value));
         }
+        if (value instanceof Instant) {
+            return LOCAL_DATE_TIME_FORMATTER.format(LocalDateTime.ofInstant((Instant) value, ZoneId.systemDefault()));
```

Comment:
`LOCAL_DATE_TIME_FORMATTER` is used for LocalDateTime and Instant whereas `Timestamp` uses `TIMESTAMP_FORMATTER`.

There are differences in how they support milli and nano seconds which is shown in the tests.

However, is this right? Trino returns true for...

```
select timestamp '2014-01-01 00:00:00' = timestamp '2014-01-01 00:00:00.000';
```

Is this the case for all databases?

Reply:
yes, this is specified in ANSI SQL. Comparision of datetimes is defined in terms of resulting interval from T1 - T2. If the interval is 0 then values compare equal. Precision doesn't matter, zones do (due to implicit conversion across zones).

Postgres and MySQL also behave the same for reference.